leshan-server-core/src/main/java/org/eclipse/leshan/server/bootstrap/BootstrapUtil.java

```diff
@@ -145,6 +179,10 @@ public static BootstrapWriteRequest toWriteRequest(int instanceId, ACLConfig acl
         for (Entry<Integer, ACLConfig> acl : bootstrapConfig.acls.entrySet()) {
             requests.add(toWriteRequest(acl.getKey(), acl.getValue(), contentFormat));
         }
+        // handle oscore //TODO: Avoid to write also bs oscore object back to client?
```

Comment:
My point here was that let's say a client communicates to the bootstrap server using OSCORE, and it gets OSCORE configuration to use towards the device manager. In such case as things are now 2 oscore objects will be written to the client. One for the bootstrap server and one for device manager. But the client already has the OSCORE security configuration to use towards the bootstrap server. So I was not sure how this should be handled (perhaps it's fine).

Reply:
Your concern makes sense but this is maybe not the right place for this TODO, because this is more a `bsserver-demo question`. I will try to explain.

There are 2 main concepts about bootstrap server.

The `SecurityStore` which contains information about how a device should connect to server (using psk, rpk , x509, or oscore ).

And `ConfigStore` which contains configuration which should be written/deleted on device during bootstrap session
Since a recent refactoring(#883), there is 2 way to create this kind of config :
- the old `BootstrapConfig` more easier to write but less flexible.
- the new `BootstrapConfiguration` which is more complex but also more flexible.

`BootstrapUtil` is just an utility class which help to go from the old way to the new.

All of those :point_up:  above is leshan concept and there is no problem here.

> the client already has the OSCORE security configuration to use towards the bootstrap server.

This is up to the user to create a config where it delete then rewrite the oscore object or do not change anything.

**But  for demo** we use a "special"(not so good?) `SecurityStore` which uses the content of  `ConfigStore` to provide securityInfo. (see [BootstrapConfigSecurityStore javadocs](https://github.com/eclipse/leshan/blob/leshan-2.0.0-M2/leshan-bsserver-demo/src/main/java/org/eclipse/leshan/server/bootstrap/demo/BootstrapConfigSecurityStore.java#L37))

In that case, in you want to allow client to connect using oscore you need a bootstrap config with oscore object and so you will write an oscore object. This is imposible to accept oscore connection and to not write an oscore object. (This is a bsserver-demo limitation only)

This works for other objects (security 0 and server 1)  because we delete it before to rewrite it.
(see [bsconfigstore.js](https://github.com/eclipse/leshan/blob/leshan-2.0.0-M2/leshan-bsserver-demo/src/main/resources/webapp/js/bsconfigstore.js#L47))

**We should probably also delete OSCORE ?** (but maybe only when we write oscore object to avoid to disturb client which does not support OSCORE object ?)

Reply:
I see, thanks for the explanation. As for deleting OSCORE, would it be good to add that to this PR also?

Reply:
or in another PR as you prefer :)

Reply:
(if you do it in another PR move the TODO)

Reply:
Thanks! I'll try to do it here already.

Reply:
I have now added this functionality in commit 8034e0f80c279666d8b4a9d0dd75048b25f306c4

I found it a bit difficult to determine if the OSCORE object is being written in a compact way. I didn't want to rely on checking hardcoded indexes in it. But perhaps there is a briefer way.

Reply:
I think it's OK.